File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const anagramsMap = {};

    for (const word of strs) {
        const sortedWord = word.split('').sort().join('');
        if (!anagramsMap[sortedWord]) {
            anagramsMap[sortedWord] = [];
        }
        anagramsMap[sortedWord].push(word);
    }

    return Object.values(anagramsMap);
};

File: 0049-group-anagrams/0049-group-anagrams.py
from collections import defaultdict
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams_map = defaultdict(list)

        for word in strs:
            sorted_word = ''.join(sorted(word))
            anagrams_map[sorted_word].append(word)

        return list(anagrams_map.values())
